boards/shared/Inc/Io/Io_SharedAdc.h

```diff
@@ -1,67 +1,55 @@
 #pragma once
-
 #include <stm32f3xx_hal.h>
-#include <arm_math.h>
-
-#include "Io_BoardSpecifics.h"
-
-// clang-format off
+#include "Io_SharedAdcConfig.h"
 
-// Memory address where you can find the internal reference voltage calibration 
-// values: Raw ADC data acquired at temperature of 30 degC with VDDA = 3.3V 
-// during the manufacturing process
-#define VREFINT_CALIBRATION_ADDRESS		(uint16_t *)(0x1FFFF7BA)
-
-// Number of microcontroller pins that are configured to be ADC inputs
-#ifndef NUM_ADC_CHANNELS
-    #error "Please define the number pins configured as ADC inputs using NUM_ADC_CHANNELS!"
+#ifndef MAX_NUM_OF_ADC
```

Comment:
Is this referring to the hardware peripheral? If so can you leave a comment describing it?
```
The number of ADC peripherals in the microcontrollers. For example, STM302F302CCT6 has two: ADC1 and ADC2.
```